python/python/bystro/ancestry/listener.py

```diff
@@ -82,16 +72,43 @@ def _get_model_from_s3(assembly: str) -> AncestryModel:
 
     logger.info("Loaded ancestry models from S3")
 
-    model = AncestryModel(pca_loadings_df, rfc)
+    return AncestryModel(pca_loadings_df, rfc)
+
+
+def _get_models_from_s3(assembly: str) -> AncestryModels:
+    if assembly in models_cache:
+        logger.info("Model for assembly %s found in cache.", assembly)
+        return models_cache[assembly]
+
+    pca_local_key_gnomad = f"{assembly}_{GNOMAD_PCA_FILE}"
+    rfc_local_key_gnomad = f"{assembly}_{GNOMAD_RFC_FILE}"
+
+    pca_file_key_gnomad = f"{assembly}/{pca_local_key_gnomad}"
+    rfc_file_key_gnomad = f"{assembly}/{rfc_local_key_gnomad}"
+
+    pca_local_key_array = f"{assembly}_{ARRAY_PCA_FILE}"
+    rfc_local_key_array = f"{assembly}_{ARRAY_RFC_FILE}"
+
+    pca_file_key_array = f"{assembly}/{pca_local_key_array}"
+    rfc_file_key_array = f"{assembly}/{rfc_local_key_array}"
+
+    gnomad_model = _get_one_model_from_s3(
+        pca_local_key_gnomad, rfc_local_key_gnomad, pca_file_key_gnomad, rfc_file_key_gnomad
+    )
+    array_model = _get_one_model_from_s3(
+        pca_local_key_array, rfc_local_key_array, pca_file_key_array, rfc_file_key_array
+    )
+
+    models = AncestryModels(gnomad_model, array_model)
 
     # Update the cache with the new model
-    if len(models_cache) >= 2:
+    if len(models_cache) >= 1:
```

Comment:
cache one assembly since we now have 2x the models per assembly, and in the future 3x